### PR TITLE
Adding language to code fences and pre to console logs

### LINKED ISSUE #40316: Code fences should have languages
This allows us to highlight them with Prism.  For contrast, https://www.freecodecamp.org/learn/scientific-computing-with-python/python-for-everybody/conditional-execution has a code language for the answers:
image
vs
https://www.freecodecamp.org/learn/scientific-computing-with-python/python-for-everybody/make-a-relational-database
image

Some challenges have console output wrapped in code fences.  Ideally these should not use code fences since they are just text - a pre element would be more semantic.

To avoid extra work, this should wait until https://github.com/freeCodeCamp/freeCodeCamp/pull/39800 is merged.

### PR DESCRIPTION
**Pull request by majestic-owl448**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #40316 

I've left a couple of questions in #40316 about what to do with the `pre` elements, those are in separate commits, so it can be reverted if needed.

### CODE REVIEW

File: curriculum/challenges/english/07-scientific-computing-with-python/python-for-everybody/dictionaries-and-loops.md [hunk 1]
```diff
@@ -26,25 +26,26 @@ for key in counts:
...
 
 ---
 
-chuck 1
+<pre>chuck 1</pre>
 
 ---
 
-\[Error]
+<pre>\[Error]</pre>
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/07-scientific-computing-with-python/python-for-everybody/dictionaries-and-loops.md:48]:
```suggestion
<pre>[Error]</pre>
```

File: curriculum/challenges/english/07-scientific-computing-with-python/python-for-everybody/reading-files.md [hunk 2]
```diff
@@ -13,23 +13,23 @@ What is used to indicate a new line in a string?
 
 ## --answers--
 
-\\n
+<pre>\\n</pre>
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/07-scientific-computing-with-python/python-for-everybody/reading-files.md:16]:
```suggestion
<pre>\n</pre>
```

File: curriculum/challenges/english/07-scientific-computing-with-python/python-for-everybody/regular-expressions-practical-applications.md [hunk 3]
```diff
@@ -13,19 +13,19 @@ What will search for a "$" in a regular expression?
 
 ## --answers--
 
-$
+<pre>$</pre>
 
 ---
 
-\\dollar\\
+<pre>\\dollar\\</pre>
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/07-scientific-computing-with-python/python-for-everybody/regular-expressions-practical-applications.md:20]:
```suggestion
<pre>\dollar\</pre>
```

File: curriculum/challenges/english/07-scientific-computing-with-python/python-for-everybody/regular-expressions-practical-applications.md [hunk 4]
```diff
@@ -13,19 +13,19 @@ What will search for a "$" in a regular expression?
...
 
 ---
 
-\\$
+<pre>\\$</pre>
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/07-scientific-computing-with-python/python-for-everybody/regular-expressions-practical-applications.md:24]:
```suggestion
<pre>\$</pre>
```

File: curriculum/challenges/english/07-scientific-computing-with-python/python-for-everybody/regular-expressions.md [hunk 5]
```diff
@@ -13,23 +13,23 @@ Which regex matches only a white space character?
 
 ## --answers--
 
-\\S
+<pre>\\S</pre>
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/07-scientific-computing-with-python/python-for-everybody/regular-expressions.md:16]:
```suggestion
<pre>\S</pre>
```

File: curriculum/challenges/english/07-scientific-computing-with-python/python-for-everybody/regular-expressions.md [hunk 6]
```diff
@@ -13,23 +13,23 @@ Which regex matches only a white space character?
...
 
 ---
 
-\\s
+<pre>\\s</pre>
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/07-scientific-computing-with-python/python-for-everybody/regular-expressions.md:20]:
```suggestion
<pre>\s</pre>
```

File: curriculum/challenges/english/07-scientific-computing-with-python/python-for-everybody/regular-expressions.md [hunk 7]
```diff
@@ -13,23 +13,23 @@ Which regex matches only a white space character?
...
 
 ---
 
-.
+<pre>.</pre>
 
 ---
 
-\_
+<pre>\_</pre>
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/07-scientific-computing-with-python/python-for-everybody/regular-expressions.md:28]:
```suggestion
<pre>_</pre>
```

File: curriculum/challenges/english/07-scientific-computing-with-python/python-for-everybody/regular-expressions.md [hunk 8]
```diff
@@ -13,23 +13,23 @@ Which regex matches only a white space character?
...
 
 ---
 
-\\.
+<pre>\\.</pre>
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/07-scientific-computing-with-python/python-for-everybody/regular-expressions.md:32]:
```suggestion
<pre>\.</pre>
```

File: curriculum/challenges/english/07-scientific-computing-with-python/python-for-everybody/variables-expressions-and-statements.md [hunk 9]
```diff
@@ -13,19 +13,19 @@ What is the symbol used in an assignment statement?
...
 
 ---
 
-=
+<pre>=</pre>
 
 ---
 
-\|
+<pre>\|</pre>
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/07-scientific-computing-with-python/python-for-everybody/variables-expressions-and-statements.md:28]:
```suggestion
<pre>|</pre>
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
In my opinion, the code blocks which had 
````
```py
```
````
Should stay. Specifically, when there are arrays (many of the numpy challenges).

What I would have changed to `pre` is just:
- Any code blocks without a language
- The tabular data

These changes are because these elements should not need escaping, when within `pre` tags.

File: curriculum/challenges/english/08-data-analysis-with-python/numpy/loading-data-and-advanced-indexing.md [hunk 10]
```diff
@@ -11,16 +11,16 @@ videoId: tUdBZ7pF8Jg
 
 Given a file named `data.txt` with these contents:
 
-<pre>
+```
```
Comment (RandellDawson) [inline curriculum/challenges/english/08-data-analysis-with-python/numpy/loading-data-and-advanced-indexing.md]: All code blocks must specify a language.  This is not code, so what is wrong with `pre` tags here?

Comment (ieahleen) [reply curriculum/challenges/english/08-data-analysis-with-python/numpy/loading-data-and-advanced-indexing.md]: It seems I lost one, adding `pre` there :+1:

**Review by ShaunSHamilton — DISMISSED** [review]
This LGTM 👍 

I am especially happy with the way the `sql` highlighting turns out.

**Comment by ojeytonwilliams** [comment]
Sorry for the delay, I should have responded when you asked in the original issue.

Regarding the `pre` tags, they're only necessary if we want to preserve spaces and newlines in the output (and are avoiding backticks because the output isn't code).  So, 

```
<pre>['csev@umich.edu', 'cwen@iupui.edu']</pre>
``` 

is not necessary, but 

```
<pre>
Tom      13
Kris     11
Ahmad     9
Beau      7
Name: Longest streak, dtype: int64
</pre>
```

is necessary.

So, multiline text in general probably wants `pre`s, since that's a lot cleaner than using `br` and `&nbsp;`.

**Comment by ieahleen** [comment]
@ojeytonwilliams should I revert the changes for the single line ones?

**Comment by ojeytonwilliams** [comment]
> @ojeytonwilliams should I revert the changes for the single line ones?

Yes, please.  That would be ideal.

File: curriculum/challenges/english/07-scientific-computing-with-python/python-for-everybody/dictionaries-and-loops.md [hunk 11]
```diff
@@ -26,25 +26,22 @@ for key in counts:
...
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/07-scientific-computing-with-python/python-for-everybody/dictionaries-and-loops.md:44]: All of the added `\` to escape need to be removed, if they are within `pre` tags:
image

Live site: https://www.freecodecamp.org/learn/scientific-computing-with-python/python-for-everybody/dictionaries-and-loops

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/07-scientific-computing-with-python/python-for-everybody/dictionaries-and-loops.md (hunk 11 above)
Comment (ieahleen) [inline curriculum/challenges/english/07-scientific-computing-with-python/python-for-everybody/dictionaries-and-loops.md:44]:
```suggestion
<pre>[Error]</pre>
```

**Review by ShaunSHamilton — APPROVED** [review]
This LGTM 👍 . Thanks, Ieahleen

**Review by moT01 — APPROVED** [review]
LGTM @ieahleen 🎉 Thanks for working on this 🎉

**Pull request merged**